dnsmasq_interface.c

```diff
@@ -909,3 +933,27 @@ int FTL_listsfile(char* filename, unsigned int index, FILE *f, int cache_size, s
 	counters.gravity += added;
 	return name_count;
 }
+
+static void block_single_domain(char *domain)
+{
+	struct all_addr addr4;
+	if(inet_pton(AF_INET, "127.0.0.1", &addr4) <= 0)
+	{
+		logg("inet_pton failed in block_single_domain(%s)!",domain);
+		return;
+	}
+
+	struct crec *cache4;
+	if((cache4 = malloc(sizeof(struct crec) + strlen(domain)+1-SMALLDNAME)))
```

Comment:
I had to check the definition of `struct crec`, but this does check out even though it seems wrong :wink:.

Reply:
`crec` has `char *namep` at its end. Hence, when we allocate enough memory for `sizeof(struct crec)` we would have only allocated space for one (!) char in this pointer. Hence, we allocate a bit more to have enough to put the domain name into. The `strlen(domain)+1-SMALLDNAME` magic is a bit more tricky, but that due to that we have a fixed char of length `SMALLDNAME` in `union name` as well.